Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -878,6 +915,59 @@ - (void)generateShortUrl:(NSArray *)tags andAlias:(NSString *)alias andType:(Bra
     });
 }
 
+- (NSString *)generateShortUrl:(NSArray *)tags andAlias:(NSString *)alias andType:(BranchLinkType)type andChannel:(NSString *)channel andFeature:(NSString *)feature andStage:(NSString *)stage andParams:(NSString *)params {
+    NSString *shortURL = nil;
+    
+    BNCServerRequest *req = [[BNCServerRequest alloc] init];
+    req.tag = REQ_TAG_GET_CUSTOM_URL;
+    BNCLinkData *post = [self prepareLinkDataFor:tags andAlias:alias andType:type andChannel:channel andFeature:feature andStage:stage andParams:params];
+    
+    if ([self.linkCache objectForKey:post]) {
+        shortURL = [self.linkCache objectForKey:post];
+    } else {
+        req.postData = post.data;
+        req.linkData = post;
+        
+        if (self.initFinished && self.hasNetwork) {
+            self.lastRequestWasInit = NO;
+            [BNCPreferenceHelper log:FILE_NAME line:LINE_NUM message:@"Created custom url synchronously"];
+            BNCServerResponse * serverResponse = [self.bServerInterface createCustomUrlSynchronous:req];
+            shortURL = [serverResponse.data objectForKey:URL];
+            
+            // cache the link
+            BNCLinkData *linkData = serverResponse.linkData;
+            if (linkData) {
+                [self.linkCache setObject:shortURL forKey:linkData];
+            }
+        } else if (self.initFailed || self.initNotCalled) {
+            NSLog(@"Branch SDK Error: making request before init succeeded!");
+        }
+    }
+    
+    return shortURL;
+}
+
+- (BNCLinkData *)prepareLinkDataFor:(NSArray *)tags andAlias:(NSString *)alias andType:(BranchLinkType)type andChannel:(NSString *)channel andFeature:(NSString *)feature andStage:(NSString *)stage andParams:(NSString *)params {
+    BNCLinkData *post = [[BNCLinkData alloc] init];
+    [post setObject:[BNCPreferenceHelper getAppKey] forKey:APP_ID];
+    [post setObject:[BNCPreferenceHelper getDeviceFingerprintID] forKey:DEVICE_FINGERPRINT_ID];
+    [post setObject:[BNCPreferenceHelper getIdentityID] forKey:IDENTITY_ID];
+    [post setObject:[BNCPreferenceHelper getSessionID] forKey:SESSION_ID];
+    
+    [post setupType:type];
+    [post setupTags:tags];
+    [post setupChannel:channel];
+    [post setupFeature:feature];
+    [post setupStage:stage];
+    [post setupAlias:alias];
+    
+    [post setObject:[NSString stringWithFormat:@"ios%@", SDK_VERSION] forKey:@"sdk"];
+    
+    NSString *args = params ? params : @"{ \"source\":\"ios\" }";
```

Comment:
Sure, either way is fine. Just wasn't sure if you were familiar with the syntax.